modules/launch_utils.py

```diff
@@ -189,7 +189,16 @@ def git_clone(url, dir, name, commithash=None):
         return
 
     try:
-        run(f'"{git}" clone "{url}" "{dir}"', f"Cloning {name} into {dir}...", f"Couldn't clone {name}", live=True)
+        if commithash is not None:
+            os.makedirs(dir)
+            now_dir = os.getcwd()
+            os.chdir(dir)
+            run(f'{git} init')
+            run(f'{git} remote add origin {url}')
+            run(f'{git} pull origin {commithash}', f"Cloning {name} into {dir}...", f"Couldn't clone {name}",live=True)

```

Comment:
`url` and `commithash` need to be quoted (preferably with `shlex.quote()`).

Reply:
url probably does but I don't think commithash can have anything that would need to be quoted

Reply:
Until it does, when a funny YouTube tutorial video says `set SOME_REPO_COMMIT='&&wget -i malicious.script,etc,etc.'` is the right thing to do to get 10,000x faster inference.

Reply:
we're quoting url with just quotes, so simple `run(f'{git} -C "{dir}" remote add origin "{url}"')` should probably be good enough.

Reply:
(Really, though, we should just use lists for these, never strings, as then `subprocess.Popen()` would always do the right thing.)

Reply:
If you'd like to change that... you're welcome to.

Reply:
some inconsistencies with how we define Python and Git env in webui-user.bat and use in webui.bat
when theres a space in path Python env will have to be define qut quote but Git don't
https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/14400#issuecomment-1867484017

Reply:
Is it good enough now?